### DISCUSSION TITLE
Non-homogeneous slip constraint

### DISCUSSION THREAD

**Discussion by johnrudge** [jorgensd/dolfinx_mpc#110] [General]
Thank you @jorgensd for all your work on the dolfin_mpc package. I was wondering if it's possible to have a non-homogeneous version of `create_slip_constraint`. At the moment the function is able to set $u \cdot v=0$ but I have a problem where the boundary condition involves setting $u \cdot v$ to a non-zero value.

**Comment by jorgensd** [comment]
I've thought about this for a while, and it isn't that hard to implement, but I've unfortunately not had time to implement it yet.
In general, I think it is a good idea to support, as it would make "DirichletBCs" in this library into MPCs with only a rhs value.

The general idea would be to add a coefficient array associated with each mpc, and use that on the rhs (in a similar way to the current lifting procedure).

**Reply by johnrudge** [reply]
Thanks for letting me know. I think I might be able to work round it for my current project, but it would be a great feature to have if you get chance to implement it. Thanks again!
